cmd/ftl/main.go

```diff
@@ -83,6 +83,9 @@ func main() {
 	kctx.BindTo(sr, (*cf.Resolver[cf.Secrets])(nil))
 	kctx.BindTo(cr, (*cf.Resolver[cf.Configuration])(nil))
 
+	config, _ := cr.LoadConfig(ctx)
```

Comment:
This is using the internal state of the ProjectConfigResolver. Instead move this function into the `projectconfig` directory as a top-level function, and pass it into ProjectConfigResolver (and Bind())

Reply:
I've now moved it to load it in separately but I'm unsure about passing it in to the resolver like we'd discussed.
Looking at the implementation, every read/write of config is loaded/written and theres no Config held in memory.

If we pass in Config to the resolvers, that would mean changing implementations so that we keep Config in memory and read from memory instead of from disk. And also seems weird to me that it would accept a Config at initialization, but when calling Set/Unset it would write to a file, which you'd hope is where the initial Config originated.

Reply:
Makes sense. Centralise the saving logic in the projectconfig package too then?